cfme/storage/volume.py

```diff
@@ -243,6 +287,31 @@ def create_snapshot(self, name, cancel=False, reset=False):
                 view.save.click()
                 return snapshot_collection.instantiate(name, self.provider)
 
+    def attach_instance(self, name, mountpoint=None, cancel=False, reset=False):
+        view = navigate_to(self, 'AttachInstance')
+        changed = view.fill({'instance': name, 'mountpoint': mountpoint})
+
+        if changed:
+            if reset:
+                view.reset.click()
+            elif cancel:
+                view.cancel.click()
+            else:
+                view.attach.click()
+
+        view.flash.assert_no_error()
+
+    def detach_instance(self, name, cancel=False):
+        view = navigate_to(self, 'DetachInstance')
+        view.instance.fill(name)
+
+        if cancel:
+            view.cancel.click()
+        else:
+            view.detach.click()
```

Comment:
Similiar comment to above, is detach only active if the view was changed? If so, this needs to be captured. 

```if changed:
        view.detach.click()```